src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -1124,6 +1124,10 @@ private void WorkspaceRemoved(WorkspaceModel item)
         {
             var viewModel = workspaces.First(x => x.Model == item);
             if (currentWorkspaceViewModel == viewModel)
+                if(currentWorkspaceViewModel != null)
+                {
+                    currentWorkspaceViewModel.Dispose();
```

Comment:
main entry point for disposal of viewModels - if we remove a workspace from the workspace collection - dispose it, this causes dispose to get called on nodeViewModel, NoteViewModel, PortViewModel, ConnectorViewModel, etc.